eclipse-theia/theia
# Add preference for visualPreview on hover

#### What it does

Made window.tabbar.enhancedPreview an enum preference with these options:
- classic: a simple unstyled preview, containing the name
- enhanced: a styled preview containing title and caption (#12350)
- visual: the enhanced preview +  visual preview of the view Extended the hover service so that it supports the visual preview. 

Added the `PreviewableWidget` interface.
Widgets, implementing this interface, can be previewed once they were loaded. The loaded flag is set to true when a widget is set to active. Widgets implementing the interface can specify how the preview should look like. The default is simply the node of the widget.
Webviews are currently not previewable, as they raise some challenges. For example, scripts in the webviews would need to be handled/blocked. Therefore, an approach for webviews should be tackled in a follow up.

Fixes #12646

Contributed on behalf of STMicroelectronics
#### How to test

1. Check out this PR an open Theia
2. Set the `window.tabbar.enhancedPreview` preference to `visual`
3. Observer, that on horizontal tab bars when hovering a tab of a view, that is currently not visible, but was loaded before, a visual preview is rendered in the hover feedback.
#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
